gymnasium/envs/tabular/blackjack.py

```diff
@@ -0,0 +1,477 @@
+            (
+                screen_width // 2 + spacing // 2,
+                dealer_text_rect.bottom + spacing,
+            ),
+        )
+
+        player_text = small_font.render("Player", True, white)
+        player_text_rect = screen.blit(
+            player_text, (spacing, dealer_card_rect.bottom + 1.5 * spacing)
+        )
+
+        large_font = get_font(os.path.join("font", "Minecraft.ttf"), screen_height // 6)
+        player_sum_text = large_font.render(str(player_sum), True, white)
+        player_sum_text_rect = screen.blit(
+            player_sum_text,
+            (
+                screen_width // 2 - player_sum_text.get_width() // 2,
+                player_text_rect.bottom + spacing,
+            ),
+        )
+
+        if usable_ace:
+            usable_ace_text = small_font.render("usable ace", True, white)
+            screen.blit(
+                usable_ace_text,
+                (
+                    screen_width // 2 - usable_ace_text.get_width() // 2,
+                    player_sum_text_rect.bottom + spacing // 2,
+                ),
+            )
+        return render_state, np.transpose(
+            np.array(pygame.surfarray.pixels3d(screen)), axes=(1, 0, 2)
+        )
+
+    def render_close(self, render_state: RenderStateType) -> None:
+        """Closes the render state."""
+        try:
+            import pygame
+        except ImportError as e:
+            raise DependencyNotInstalled(
+                "pygame is not installed, run `pip install gymnasium[classic_control]`"
+            ) from e
+        pygame.display.quit()
+        pygame.quit()
+
+
+class BlackJackJaxEnv(FunctionalJaxEnv, EzPickle):
+    """A Gymnasium Env wrapper for the functional blackjack env."""
+
+    metadata = {"render_modes": ["rgb_array"], "render_fps": 50}
+
+    def __init__(self, render_mode: Optional[str] = None, **kwargs):
+        """Initializes Gym wrapper for blackjack functional env."""
+        EzPickle.__init__(self, render_mode=render_mode, **kwargs)
+        env = BlackjackFunctional(**kwargs)
+        env.transform(jax.jit)
+
+        super().__init__(
+            env,
+            metadata=self.metadata,
+            render_mode=render_mode,
+        )
+
+
+# Pixel art from Mariia Khmelnytska (https://www.123rf.com/photo_104453049_stock-vector-pixel-art-playing-cards-standart-deck-vector-set.html)
+
+# Jax structure inspired by https://medium.com/@ngoodger_7766/writing-an-rl-environment-in-jax-9f74338898ba
+
+
+if __name__ == "__main__":
+    """
+    Temporary environment tester function.
+    """
+
+    env = HumanRendering(BlackJackJaxEnv(render_mode="rgb_array"))
+
+    obs, info = env.reset()
+    print(obs, info)
+    env.render()
+
+    terminal = False
+    while not terminal:
+        action = int(input("Please input an action\n"))
+        obs, reward, terminal, truncated, info = env.step(action)
+        print(obs, reward, terminal, truncated, info)
+        env.render()
```

Comment:
This is not actually necessary. Technically, on each `step` if render_mode="human" then `render` is called within the environment

Reply:
removed